yew/src/scheduler.rs

```diff
@@ -97,6 +98,10 @@ impl Scheduler {
     }
 
     pub(crate) fn start(&self) {
+        // The lock here is used to prevent recursion. It is assumed
+        // that if the lock here fails to acquire, it is because this
```

Comment:
I think you should use stronger language than just 'It is assumed ..'.
There is no other place where the lock is acquired and Rust guarantees that it only stays borrowed for the duration of `start`.